src/main/java/org/opentripplanner/middleware/trip_monitor/jobs/CheckMonitoredTrip.java

```diff
@@ -0,0 +1,391 @@
+        checkSkipped = shouldSkipMonitoredTripCheck(trip);
+        if (checkSkipped) {
+            LOG.debug("Skipping check for trip: {}", trip.id);
+            return;
+        }
+        // Make a request to OTP with the monitored trip params or if this is in a test environment use the injected OTP
+        // response.
+        OtpDispatcherResponse otpDispatcherResponse = null;
+        try {
+            otpDispatcherResponse = injectedOtpResponseForTesting != null
+                ? injectedOtpResponseForTesting
+                : OtpDispatcher.sendOtpPlanRequest(generateTripPlanQueryParams(trip));
+        } catch (Exception e) {
+            // TODO: report bugsnag
+            LOG.error("Encountered an error while making a request ot the OTP server. error={}", e);
+            return;
+        }
+
+        if (otpDispatcherResponse.statusCode >= 400) {
+            // TODO: report bugsnag
+            LOG.error("Received an error from the OTP server. status={}", otpDispatcherResponse.statusCode);
+            return;
+        }
+        otpResponse = otpDispatcherResponse.getResponse();
+        // Check monitored trip.
+        runCheckLogic();
+        // Send notifications to user. This should happen before updating the journey state so that we can check the
+        // last notification sent.
+        sendNotifications();
+        // Update journey state.
+        JourneyState journeyState = trip.retrieveJourneyState();
+        journeyState.update(this);
+    }
+
+    private void runCheckLogic() {
+        // TODO: Should null tripRequestId be fixed?
+        // TripSummary tripSummary = new TripSummary(otpDispatcherResponse.response.plan, otpDispatcherResponse.response.error, null);
+        matchingItineraryIndex = findMatchingItineraryIndex(trip, otpResponse);
+        if (matchingItineraryIndex == -1) {
+            // No matching itinerary was found.
+            enqueueNotification(TripMonitorNotification.createItineraryNotFoundNotification());
+            // TODO: Try some fancy things to construct a matching itinerary (e.g., transit index?).
+            return;
+        }
+        // Matching itinerary found in OTP response. Run real-time checks.
+        Itinerary itinerary = otpResponse.plan.itineraries.get(matchingItineraryIndex);
+        enqueueNotification(
+            // Check for notifications related to service alerts.
+            checkTripForNewAlerts(trip, itinerary),
+            // Check for notifications related to delays.
+            checkTripForDepartureDelay(trip, itinerary),
+            checkTripForArrivalDelay(trip, itinerary)
+        );
+    }
+
+    /**
+     * Find an itinerary from the OTP response that matches the monitored trip's stored itinerary.
+     */
+    private static int findMatchingItineraryIndex(MonitoredTrip trip, Response otpResponse) {
+        for (int i = 0; i < otpResponse.plan.itineraries.size(); i++) {
```

Comment:
`otpResponse.plan` should be checked for `null`. There is no guarantee OTP will return a plan (e.g. when attempting to check a weekday trip during the weekend when equivalent service is not offered).

Reply:
To be addressed in #70.